clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -517,7 +517,7 @@ void pollSelectionKeys(Set<SelectionKey> selectionKeys,
 
             // register all per-connection metrics at once
             sensors.maybeRegisterConnectionMetrics(nodeId);
-            if (idleExpiryManager != null)
+            if (idleExpiryManager != null && !explicitlyMutedChannels.contains(channel))
```

Comment:
This is limiting the times when we perform the update. For example when sending response, the channel is active, but explicitly muted. Don't think we want to skip update in this case.